src/Build/BackEnd/Components/Logging/LoggingService.cs

```diff
@@ -1592,6 +1593,12 @@ private void RouteBuildEvent(object loggingEvent)
                 _buildSubmissionIdsThatHaveLoggedErrors.Add(errorEvent.BuildEventContext?.SubmissionId ?? BuildEventContext.InvalidSubmissionId);
             }
 
+            if (buildEventArgs is BuildCheckResultError checkResultError)
```

Comment:
We should probably handle the warning + ShouldTreatWarningAsError as well

Reply:
So there is one condition build with -err:BC0101

Reply:
Updated with new fix

Reply:
@JanKrivanek please help review